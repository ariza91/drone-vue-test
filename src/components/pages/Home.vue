<template>
  <div class="home">
    <router-link :to="{ name: 'DroneList' }">
      <button class="btn btn-outline-primary">
        {{ $t("Home.droneList") }}
      </button>
    </router-link>
  </div>
</template>

<script>
export default {
  name: "Home",
};
</script>

<style scoped lang="scss">
@import "@/style/theme.scss";

.home {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 15%;

  img {
    width: 50%;
  }
}
</style>