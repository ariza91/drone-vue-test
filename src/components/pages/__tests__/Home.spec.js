import { shallowMount } from "@vue/test-utils";
import Home from "@/components/pages/Home.vue";

const factory = () => shallowMount(Home, {
    mocks: { $t: (value) => value },
    stubs: ['router-link']
});

describe("Home", () => {
    let cmp;

    beforeEach(() => {
        cmp = factory();
    })

    describe('renders', () => {
        it("renders correctly", () => {
            expect(cmp).toMatchSnapshot()
        });
    })
});
