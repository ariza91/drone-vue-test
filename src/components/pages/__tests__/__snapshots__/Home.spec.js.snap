// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders renders correctly 1`] = `
<div class="home">
  <router-link-stub to="[object Object]"><button class="btn btn-outline-primary">
      Home.droneList
    </button></router-link-stub>
</div>
`;
